src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py

```diff
@@ -91,25 +93,39 @@ def uploadWorker(workInput, results, dbsUrl, gzipEncoding=False):
             logging.info("About to call insert block for: %s", name)
             dbsApi.insertBulkBlock(blockDump=block)
             results.put({'name': name, 'success': "uploaded"})
-        except Exception as ex:
-            exString = str(getattr(ex, "body", ex))
-            if 'Block %s already exists' % name in exString:
+        except HTTPError as ex:
+            # when we will have new dbs3-client and fine granularity DBS Go server
+            # error codes then we should remove if/elif/else and replace it with
+            # DBSError(ex.body), e.g.
+            # dbsError = DBSError(ex.body)
+            # reason = dbsError.getReason()
+            # srvCode = dbsError.getServverCode()
+            # results.put({'name': name, 'success': 'error', 'error': reason, 'server_code': srvCode})
+
+            # legacy code which deal with DBS python server exceptions
+            dbsError = DBSError(ex.body)
+            reason = dbsError.getReason()
+            if 'Block %s already exists' % name in reason:
```

Comment:
Valentin, can you please confirm which HTTPError error field this message (Block BLAH already exists) is supposed to be propagated? Is it `reason` or `message`?